app/services/foreman_deprecation.rb

```diff
@@ -0,0 +1,24 @@
+class ForemanDeprecation
+  def self.deprecation_warning(moved_to_controller, moved_to_method, foreman_version_deadline, additional_info = nil,callstack = caller)
+    raise Foreman::Exception, "Invalid version format, please enter in x.y.z" unless foreman_version_deadline.to_s.match(/^\d[.]\d\d?[.]\d$/)
```

Comment:
1. extract all strings passed to Foreman::Exception
1. use \A and \z in regexes to test for string termination
1. nitpick, use \d+ instead of \d\d?
1. perhaps use Gem instead of a regexp?